src/state/README.md

```diff
@@ -52,3 +52,103 @@ app.container.on('connected', async () => {
     await UserData.delete('my-value');
 });
 ```
+### IndexedDB
+Any work with indexedDB or the userData plugin as a whole should be done inside the 'connected' callback method
+
+``` javascript
+import { UserData } from 'springroll';
+
+app.container.on('connected', async () => {
+  // connect to your database
+  let response = await userData.OpenDb('dbName');
+  
+  // Work with IndexedDB below
+  ...
+}
+```
+
+In order to  change the structure of the database, such as adding/removing stores to a database or an index to a store, you should do so in the IDBOpen method
+
+``` javascript
+  
+  // Additions is an optional parameter expecting a JSON object with any additions to the databases structure namely new stores and indexes. These are placed inside of an array 
+  let additions = {
+    stores: [{
+      storeName: 'storeOne',
+      // optionally define a keyPath and/or set autoIncrement to true or false
+      options: { keyPath: "taskTitle" }
+    },
+    {
+      storeName: 'storeTwo'
+    }],
+    indexes: [{
+      indexName: 'newIndex',
+      keyPath: 'key',
+      // Any objectParameters for the Index can be placed here
```

Comment:
similar to my comment about the MDN docs, anywhere where we have something like an options object we should link out to a page that explains it a little more in depth. In case users are unfamiliar. That way we also don't need to worry about explaining it ourselves.